src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -550,34 +568,49 @@ export class ReplaceAction extends AbstractSearchAndReplaceAction {
 	override async run(): Promise<any> {
 		this.enabled = false;
 
-		await this.element.parent().replace(this.element);
-		const elementToFocus = this.getElementToFocusAfterReplace();
+		const elementsToReplace = getElementsToOperateOn(this.viewer, this.element);
+		let currentBottomFocusElement;
+
+		if (elementsToReplace.usingMultiselect) {
+			currentBottomFocusElement = <Match>elementsToReplace.result[elementsToReplace.result.length - 1];
+		} else {
+			currentBottomFocusElement = <Match>this.viewer.getFocus()[0];
+		}
+
+		const elementToFocus = elementIsEqualOrParent(currentBottomFocusElement, this.element) ? this.getElementToFocusAfterReplace(currentBottomFocusElement) : null;
+
+		elementsToReplace.result.forEach((elem) =>
+			elem.parent().replace(<any>elem)
```

Comment:
Cast to `any` still needed?

Reply:
https://github.com/microsoft/vscode/pull/154847#discussion_r921463556